controller_interface/include/controller_interface/controller_interface_base.hpp

```diff
@@ -200,8 +201,22 @@ class ControllerInterfaceBase : public rclcpp_lifecycle::node_interfaces::Lifecy
 
   std::shared_ptr<const rclcpp_lifecycle::LifecycleNode> get_node() const;
 
+  /**
+   * Get the current lifecycle state of the controller node.
+   * \note This method is not real-time safe and should not be called in the control loop.
```

Comment:
```suggestion
   * \note Accessing members of the returned rclcpp_lifecycle::State is not real-time safe and should not be called in the control loop.
```
nitpick, but it helped me to understand the actual issue